Sources/Sentry/SentryDefaultRateLimits.m

```diff
@@ -1,106 +1,97 @@
- * 60 seconds (default, see `defaultRadioSilenceDeadline`).
- */
-- (void)update:(NSHTTPURLResponse *)response {
-    self.radioSilenceDeadline = [self parseRetryAfterHeader:response.allHeaderFields[@"Retry-After"]];
-}
-
-/**
- * used if actual time/deadline couldn't be determined.
- */
-- (NSDate *)defaultRadioSilenceDeadline {
-    return [[SentryCurrentDate date] dateByAddingTimeInterval:60];
+- (BOOL)isRetryAfterHeaderLimitActive {
+    if (nil == self.retryAfterHeaderDate) {
+        return NO;
+    }
+    
+    BOOL isActive = [self isInFuture:self.retryAfterHeaderDate];
+    if (isActive) {
+        [SentryLog logWithMessage:[NSString stringWithFormat:@"Retry-After limit reached until: %@",  self.retryAfterHeaderDate] andLevel:kSentryLogLevelDebug];
+        return YES;
+    } else {
+        return NO;
+    }
 }
 
-/**
- * parses value of HTTP Header "Retry-After" which in most cases is sent in
- * combination with HTTP status 429 Too Many Requests.
- *
- * Retry-After value is a time-delta in seconds or a date.
- * In every case this method computes the date aka. `radioSilenceDeadline`.
- *
- * See RFC2616 for details on "Retry-After".
- * https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.37
- *
- * @return NSDate representation of Retry-After.
- *         As fallback `defaultRadioSilenceDeadline` is returned if parsing was
- *         unsuccessful.
- */
-- (NSDate *)parseRetryAfterHeader:(NSString * __nullable)retryAfterHeader {
-    if (nil == retryAfterHeader || 0 == [retryAfterHeader length]) {
-        return [self defaultRadioSilenceDeadline];
+- (BOOL)isInFuture:(NSDate *)date {
+    NSComparisonResult result = [[SentryCurrentDate date] compare:date];
+    if (result == NSOrderedAscending) {
+        return YES;
     }
+    
+    return NO;
+}
 
-    NSDate *now = [SentryCurrentDate date];
-
-    // try to parse as double/seconds
-    double retryAfterSeconds = [retryAfterHeader doubleValue];
-    NSLog(@"parseRetryAfterHeader string '%@' to double: %f", retryAfterHeader, retryAfterSeconds);
-    if (0 != retryAfterSeconds) {
-        return [now dateByAddingTimeInterval:retryAfterSeconds];
+- (void)update:(NSHTTPURLResponse *)response {
+    NSDate* retryAfterHeaderDate = nil;
+    if (response.statusCode == 429) {
+        retryAfterHeaderDate = [self.retryAfterHeaderParser parse:response.allHeaderFields[@"Retry-After"]];
+        
+        if (nil == retryAfterHeaderDate) {
+            // parsing failed use default value
+            retryAfterHeaderDate = [[SentryCurrentDate date] dateByAddingTimeInterval:60];
+        }
     }
-
-    // parsing as double/seconds failed, try to parse as date
-    NSDate *retryAfterDate = [self.httpDateParser dateFromString:retryAfterHeader];
-
-    if (nil == retryAfterDate) {
-        // parsing as seconds and date failed
-        return [self defaultRadioSilenceDeadline];
+    
+    NSString *rateLimitsHeader = response.allHeaderFields[@"X-Sentry-Rate-Limits"];
```

Comment:
I suggest we move this first, and in case we find the header, we parse and don't bother with the `429` bits.
This new header is a replacement of the `429`/`Retry-After`. 

The instance created to handle it would ideally be lazily instantiated (i.e pass a function that gives an instance) and only invoke in the first time we handle a `429` which would only be when hitting on-premise Sentry that hasn't updated to latest version yet. Meaning this is be only a fallback for old on-prem.

Reply:
The `429` is an official HTTP response status code standing for [Too Many Requests](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/429). The [`Retry-After`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After) header is also an official HTTP header, which could also be sent with a [`503`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/503) or [`301`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/301). The `X-Sentry-Rate-Limits` is a custom header and according to our internal documentation on [Billing and Rate Limiting](https://www.notion.so/sentry/Billing-and-Rate-Limiting-b5fb092806c9469bb6e4ed7927f46045) it can be returned from any response. The example there even shows a `429` with `X-Sentry-Rate-Limits` 
```
HTTP/1.1 429 Too Many Requests
Retry-After: 2700
X-Sentry-Rate-Limits: 
  60:transaction:key, 
  2700:default;event;security:organization
```
On [Android](https://github.com/getsentry/sentry-android/blob/master/sentry-core/src/main/java/io/sentry/core/transport/HttpTransport.java#L307) for example, we parse first the `X-Sentry-Rate-Limits` and if this header is not set we check if it's a `429` with `Retry-After`, like you pointed it out. This doesn't match with what we have in our internal documentation. So either we have to update the docs or change the implementation. Do you know which one is correct? Maybe @marandaneto you know more on this topic?

Even if our own Sentry servers don't use `429` anymore, maybe it makes sense to keep it, because it's an official HTTP response status code?

Reply:
we should parse `X-Sentry-Rate-Limits` for all the responses, even 200.
if it's not available, check the `errorCode`, if it's `429`, then parse `Retry-After`
cc @jan-auer

Reply:
the idea of doing this was -> `Can be added to any response, even 200 responses. This can be useful to proactively inform SDKs that certain event-types are disabled before they even try to send it.`

Reply:
> we should parse `X-Sentry-Rate-Limits` for all the responses, even 200.
> if it's not available, check the `errorCode`, if it's `429`, then parse `Retry-After`
> cc @jan-auer

Thanks for the information @marandaneto. Still I am not 100% sure if I get this right. According to the statement above in the following response `Retry-After: 1000` is completely ignored because we have a `X-Sentry-Rate-Limits` header. Is this correct?
```
HTTP/1.1 429 Too Many Requests
Retry-After: 1000
X-Sentry-Rate-Limits: 
  60:transaction:key, 
  100:default;event;security:organization
```

Reply:
@philipphofmann yes, `Retry-After` is the fallback to keep compatibility with older Sentry versions. one may use the new SDK and the old Sentry Server which doesn't know about `X-Sentry-Rate-Limits`

Reply:
Wouldn't this kind of violate the 429 standard? It states that it may include a Retry-After header. In the scenario above it does and we would ignore it.

Reply:
@philipphofmann well, if you want to honor the 429-RFC yes, but you wouldn't be able to return the rate limits for more than one item too. so I'd say it's fine because the idea here is to have a single header that gives us all the information we need to proper rate limit on the client.
I believe `Retry-After` would be only useful for people that have their own custom transport, ideally, all of our SDKs will only look at `X-Sentry-Rate-Limits` anyway.

Reply:
so tldr, `X-Sentry-Rate-Limits` is the new `Retry-After`, 429ish

Reply:
Thanks for the clarification @marandaneto 👏

Reply:
@philipphofmann you're welcome, and not sure about removing it, as I said, that would be a breaking change for custom transports out there in the wild